packages/browser-wallet/src/popup/popupX/pages/EarningRewards/Delegator/TransactionFlow.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable react/destructuring-assignment */
+import React, { useCallback, useState } from 'react';
+import { Location, Navigate, useLocation, useNavigate } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { AccountInfoType, DelegationTargetType } from '@concordium/web-sdk';
+
+import { absoluteRoutes } from '@popup/popupX/constants/routes';
+import MultiStepForm from '@popup/shared/MultiStepForm';
+import { useSelectedAccountInfo } from '@popup/shared/AccountInfoListenerContext/AccountInfoListenerContext';
+import { formatCcdAmount } from '@popup/popupX/shared/utils/helpers';
+import FullscreenNotice, { FullscreenNoticeProps } from '@popup/popupX/shared/FullscreenNotice';
+import Page from '@popup/popupX/shared/Page';
+import Button from '@popup/popupX/shared/Button';
+import DelegatorStake from './Stake';
+import DelegatorType from './Type';
+import { configureDelegatorPayloadFromForm, type DelegatorForm } from './util';
+import { DelegationResultLocationState } from './Result/DelegationResult';
+
+function NoChangesNotice(props: FullscreenNoticeProps) {
+    const { t } = useTranslation('x', { keyPrefix: 'earn.delegator.update.noChangesNotice' });
+    return (
+        <FullscreenNotice {...props}>
```

Comment:
**Question (non-blocking):** I find the flow hard to follow since some of it is tracked in the path and some in modals, why is this even a modal and not just another page part of the routing? :smile:

Reply:
The point of these "notices" is to have the user confirm some action or simply show some information to the user. Routing to another page tears down the state built for that route/page, which would require you to save the state somehow to allow the user to return to the previous page and rebuild the page from the saved state.

We can completely avoid this by treating this as what it actually is (a prompt for the user) instead of what it looks like (a subsequential page in the flow) 😄